lib/cluster.js

```diff
@@ -89,56 +115,89 @@ Cluster.prototype.fetchStats = function fetchStats(callback) {
+        onComplete(null, allStats);
+    };
 
-        if (!stats.membership) {
-            callback(new Error('stats did not contain membership'));
-            return;
+    this.getSeedList(function (err, seeds) {
+        if (err) {
+            return callback(err);
         }
+        queueMembers(seeds);
+    });
 
-        if (!stats.membership.members) {
-            callback(new Error('membership did not contain members'));
-            return;
-        }
+    return;
 
-        async.mapLimit(stats.membership.members, self.fetchLimit, mapMember, onComplete);
+    function mapMember(memberAddr, next) {
+        self.adminClient.stats(memberAddr, function(err, stats) {
+            var statsObj = new Stats();
+            statsObj.address = memberAddr;
+
+            if (err) {
+                next(null, statsObj);
+                return;
+            }
+
+            if (!stats) {
+                callback(new Error('stats could not be gathered'));
+                return;
+            }
+
+            if (!stats.membership) {
+                callback(new Error('stats did not contain membership'));
+                return;
+            }
+
+            if (!stats.membership.members) {
+                callback(new Error('membership did not contain members'));
+                return;
+            }
+
+            statsObj.membershipChecksum = stats.membership.checksum;
+            statsObj.members = stats.membership.members;
+            statsObj.node = memberAddr;
+
+            // queue this members for query
+            queueMembers(
+                stats.membership.members.map(
+                    function getAddressFromMember(member) {
+                        return member.address;
+                    }
+                )
+            );
+
+            next(null, statsObj);
+        });
+    }
 
-        function mapMember(member, next) {
-            self.adminClient.stats(member.address, function(err, stats) {
-                var statsObj = new Stats();
-                statsObj.address = member.address;
+    function onComplete(err, allStats) {
+        self.lastDownloadTime = Date.now() - downloadTime;
+        self.allStats = allStats;
+        self.allStats.forEach(function eachStats(stats) {
+            self.parseStats(stats);
+        });
 
-                if (err) {
-                    next(null, statsObj);
-                    return;
-                }
+        callback();
+    }
 
-                statsObj.membershipChecksum = stats.membership.checksum;
-                statsObj.members = stats.membership.members;
+    function queueMembers(membersAddr) {
+        membersAddr.forEach(function (memberAddr) {
+            if (memberAddr in foundMembers) {
+                return;
+            }
 
-                next(null, statsObj);
-            });
-        }
+            foundMembers[memberAddr] = true;
+            fetchQueue.push(memberAddr, function (err, stats) {
+                if (err) return;
```

Comment:
It could be the case that fetching stats from the member errors out and we have no record of a member even though we made an attempt. Ideally we should strive to make ringpop-admin resilient to errors, transparent about them occurring, and represent them visually somehow. ringpop-admin hasn't been great about that before.. not sure if you want to start with this PR either. But keep it in mind for next time.